src/rest-devices.c

```diff
@@ -0,0 +1,572 @@
+/*
+ * Punica - LwM2M server with REST API
+ * Copyright (C) 2018 8devices
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <string.h>
+
+#include "restserver.h"
+#include "rest-list.h"
+#include "settings.h"
+
+static void rest_devices_clean_entry(database_entry_t *entry)
+{
+    if (entry->uuid)
+    {
+        free(entry->uuid);
+    }
+    if (entry->psk)
+    {
+        free(entry->psk);
+    }
+    if (entry->psk_id)
+    {
+        free(entry->psk_id);
+    }
+}
+
+static void rest_devices_delete_list(rest_list_t *list, database_entry_t *data)
```

Comment:
The `rest_` prefix seems to be a common style among all functions dealing with rest things (including callbacks and static functions used by said callbacks), not just those for /devices. We could move this discussion to https://github.com/punica/punica/pull/35 since this would need more changes.